cs50/2017/fall/vigenere/check50/__init__.py

```diff
@@ -39,6 +39,16 @@ def bazworld(self):
         self.spawn("./vigenere baz").stdin("world!$?").stdout("ciphertext:\s*xoqmd!\$\?\n", "ciphertext: xoqmd!$?\n").exit(0)
 
     @check("compiles")
+    def CSFiftyBrianyu28(self):
+        """encrypts "Brianyu28" as "Djnisrs28" using "CSFifty" as keyword"""
+        self.spawn("./vigenere CSFifty").stdin("Brianyu28").stdout("ciphertext:\s*Djnisrs28\n", "ciphertext: Djnisrs28\n").exit(0)
+
+    @check("compiles")
+    def ThisisCSFiftyMilesDanielFertel(self):
+        """encrypts "Miles Daniel Fertel" as "Fptwa Fssqje Ylzlmd" using "ThisisCSFifty" as keyword"""
+        self.spawn("./vigenere ThisisCSFifty").stdin("Miles Daniel Fertel").stdout("ciphertext:\s*Fptwa Fssqje Ylzlmd\n", "ciphertext: Fptwa Fssqje Ylzlmd\n").exit(0)
+
+    @check("compiles")
```

Comment:
What are these additional checks checking that aren't covered by existing checks?

Reply:
I had a student get a 5/5 on correctness on vigenere by hardcoding in a key length of 3, because we only test keys of length 3 for all input sizes greater than the length of the key. These two tests provide coverage for that, as well as coverage for an input that is a phrase rather than a single word.